### ISSUE TITLE
Libvirt / KVM support

### ISSUE THREAD

**Issue by alex27riva** [Orange-Cyberdefense/GOAD#237]
Hi everyone, I wanted to know if Libvirt support can be added for this project.
I have a Proxmox machine but I don't have enough resources to run an AD environment.
I want to deploy this on my PC with Fedora Workstation 40, where I use [libvirt / KVM ](https://docs.fedoraproject.org/en-US/quick-docs/virtualization-getting-started/)for virtualization.
KVM is more efficient than type 2 hypervisors like VirtualBox.

**Comment by noraj** [comment]
This shouldn't require much effort, since there is already a robust libvirt plugin for Vagrant:

https://github.com/vagrant-libvirt/vagrant-libvirt

**Comment by alex27riva** [comment]
> This shouldn't require much effort, since there is already a robust libvirt plugin for Vagrant:
> 
> https://github.com/vagrant-libvirt/vagrant-libvirt

Yes, I tried implementing it but the [StefanScherer/windows_2019](https://app.vagrantup.com/StefanScherer/boxes/windows_2019) doesn't support the libvirt provider.

**Comment by noraj** [comment]
> This shouldn't require much effort, since there is already a robust libvirt plugin for Vagrant:

Modifying goad.sh for that is easy. But creating a `Vagrantfile` and `inventory` files for libvirt provider for each lab takes a bit more time.

> Yes, I tried implementing it but the [StefanScherer/windows_2019](https://app.vagrantup.com/StefanScherer/boxes/windows_2019) doesn't support the libvirt provider.

Maybe another image could be used for libvirt?

![image](https://github.com/Orange-Cyberdefense/GOAD/assets/16578570/c4b251dd-4795-45d9-a147-7f15a142926a)

**Comment by Mayfly277** [comment]
could be easy to do with goad v3 as i delete a lot of duplicate code and do a global template by provider.
with vagrant support it should be pretty easy to do.

**Comment by andreashappe** [comment]
any suggestions how to best start with this? I am currently using goadv3 with virtualbox on a linux host and am experiencing 'weird' network problems which I believe are mostly due to virtualbox.

**Comment by 189er** [comment]
I also like Libvirt

**Comment by 189er** [comment]
Ludus :Based on Proxmox (KVM/LXC)

**Comment by noraj** [comment]
Does it work with libvirt + KVM + vagrant-libvirt? (not Proxmox or Ludus)

It's not officially listed at https://orange-cyberdefense.github.io/GOAD/providers/

@alex27riva Could you re-open the issue please.
